src/basic/path-util.c

```diff
@@ -373,52 +372,6 @@ char *path_simplify(char *path) {
         return path;
 }
 
-int path_simplify_and_warn(
-                char *path,
-                unsigned flag,
-                const char *unit,
-                const char *filename,
-                unsigned line,
-                const char *lvalue) {
```

Comment:
not sure I follow, but why is src/basic/ not linked into libsystemd-shared, too?

Reply:
`libsystemd-basic.a` obviously _is_ linked into `libsystemd-shared.so`. In fact, we use `link_whole` in that particular case, so all of `libsystemd-basic.a` will be copied into `libsystemd-shared.so`. So for anything that links to `libsystemd-shared.so`, there should be no difference, because we end up with the object in the library anyway. But other things which use `libsystemd-basic.a` but don't link to `-shared`, could end up with a copy too. I *think* it should be removed when in some optimization pass, but, as shown by the grep that is included in the commit message, this doesn't (always?) happen.

In general, if we can move things from `basic/` to `shared/`, this is always strictly better. It reduces the dependency scope of `basic/`, so it makes partial compilations faster. In some cases it also reduces the size and dependencies of things that don't link to `libsystemd-shared.so`. (Not in this case though.)